features/admin.roles.v2/components/edit-role/edit-role.tsx

```diff
@@ -80,30 +81,75 @@ export const EditRole: FunctionComponent<EditRoleProps> = (props: EditRoleProps)
+        (state: AppState) => state?.config?.ui?.features?.userRolesV3);
     const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
     const administratorRoleDisplayName: string = useSelector(
         (state: AppState) => state?.config?.ui?.administratorRoleDisplayName);
     const userRolesDisabledFeatures: string[] = useSelector((state: AppState) => {
         return state.config.ui.features?.userRoles?.disabledFeatures;
     });
+    const enableScim2RolesV3Api: boolean = useSelector(
+        (state: AppState) => state.config.ui.enableScim2RolesV3Api
+    );
+
     const isSharedRole: boolean = useMemo(() => roleObject?.properties?.some(
         (property: RolePropertyInterface) =>
             property?.name === LocalRoleConstants.IS_SHARED_ROLE && property?.value === "true"), [ roleObject ]);
 
     const isReadOnly: boolean = useMemo(() => {
+        if (enableScim2RolesV3Api) {
+            return !isFeatureEnabled(
+                featureConfig,
+                LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE"))
+                || !hasRequiredScopes(rolesV3featureConfig,
+                    rolesV3featureConfig?.scopes?.update, allowedScopes)
+                || roleObject?.meta?.systemRole;
+        } else {
+            return !isFeatureEnabled(
+                featureConfig,
+                LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE"))
+                || !hasRequiredScopes(featureConfig,
+                    featureConfig?.scopes?.update, allowedScopes)
+                || roleObject?.meta?.systemRole;
+        }
+    }, [ enableScim2RolesV3Api, featureConfig, rolesV3featureConfig, allowedScopes, roleObject ]);
+
+    const isGroupReadOnly: boolean = useMemo(() => {
+
+        if (enableScim2RolesV3Api) {
+            return !isFeatureEnabled(featureConfig,
+                LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE"))
+            || !AuthenticateUtils.hasScope("internal_role_mgt_groups_update", allowedScopes)
+            || roleObject?.meta?.systemRole;
+        }
+
         return !isFeatureEnabled(featureConfig,
             LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE"))
             || !hasRequiredScopes(featureConfig,
-                featureConfig?.scopes?.update, allowedScopes)
+                featureConfig?.scopes?.update,
+                allowedScopes)
             || roleObject?.meta?.systemRole;
-    }, [ featureConfig, allowedScopes ]);
+    }, [ enableScim2RolesV3Api, featureConfig, allowedScopes, roleObject ]);
 
     const isUserReadOnly: boolean = useMemo(() => {
-        return !isFeatureEnabled(usersFeatureConfig,
+        if (enableScim2RolesV3Api) {
+            return !isFeatureEnabled(featureConfig,
+                LocalRoleConstants.FEATURE_DICTIONARY.get("ROLE_UPDATE")) ||
+                !AuthenticateUtils.hasScope("internal_role_mgt_users_update", allowedScopes);
```

Comment:
shall we get the scope value through a constant?

Reply:
Fixed with: https://github.com/wso2/identity-apps/pull/8184/commits/364f88e7a85aa677d870962f791af4ab219be515